src/Driver/RedisSetDriver.php

```diff
@@ -120,14 +123,16 @@ public function wait(Closure $callback): void
                 $messageString = $this->redis->sPop($this->key);
             }
 
-            if (!$messageString) {
+            if ($messageString) {
+                $message = $this->serializer->unserialize($messageString);
+                $callback($message);
+                $this->incrementProcessedItems();
+            } else {
                 if ($this->refreshInterval) {
+                    $this->checkRestart();
```

Comment:
This is inconsistent with `AmazonSqsDriver.php`. `$this->refreshInterval` is there checked also if message is processed. Is there some reason why redis doesn't sleep after processing message and amazon does?

_// We are not using amazon implementation, so I don't care :) Just noticed it._

Reply:
thx, fixed